contracts/clients-tests/BlsProvider.test.ts

```diff
@@ -0,0 +1,535 @@
+    expect(spy).to.have.been.called.twice;
+  });
+
+  it("should return failures as a json string and throw an error when sending an invalid transaction", async () => {
+    // Arrange
+    const invalidEthValue = parseEther("-1");
+
+    const unsignedTransaction = {
+      ethValue: invalidEthValue,
+      contractAddress: signers[1].address,
+      encodedFunction: "0x",
+    };
+    const signedTransaction = await blsSigner.signBlsTransaction(
+      unsignedTransaction,
+    );
+
+    // Act
+    const result = async () =>
+      await blsProvider.sendBlsTransaction(signedTransaction, blsSigner);
+
+    // Assert
+    await expect(result()).to.be.rejectedWith(
+      Error,
+      '[{"type":"invalid-format","description":"field operations: element 0: field actions: element 0: field ethValue: hex string: missing 0x prefix"},{"type":"invalid-format","description":"field operations: element 0: field actions: element 0: field ethValue: hex string: incorrect byte length: 8.5"}]',
+    );
+  });
+
+  it("should retrieve a transaction receipt given a valid hash", async () => {
+    // Arrange
+    const recipient = signers[1].address;
+    const transactionResponse = await blsSigner.sendTransaction({
+      to: recipient,
+      value: parseEther("1"),
+    });
+
+    // Act
+    const transactionReceipt = await blsProvider.getTransactionReceipt(
+      transactionResponse.hash,
+    );
+
+    // Assert
+    // TODO: bls-wallet #412 Update values returned in bundle receipt to more closely match ethers transaction response
+    expect(transactionReceipt).to.be.an("object");
+    expect(transactionReceipt).to.have.property("to").to.equal("0x");
+    expect(transactionReceipt).to.have.property("from").to.equal("0x");
+    expect(transactionReceipt)
+      .to.have.property("contractAddress")
+      .to.equal("0x");
+    expect(transactionReceipt).to.have.property("transactionIndex"); // .to.equal("0x");
+    expect(transactionReceipt)
+      .to.have.property("gasUsed")
+      .to.equal(parseEther("0"));
+    expect(transactionReceipt).to.have.property("logsBloom").to.equal("");
+    expect(transactionReceipt).to.have.property("blockHash"); // .to.equal("0x");
+    expect(transactionReceipt).to.have.property("transactionHash"); // .to.equal("0x");
+    expect(transactionReceipt).to.have.property("logs").to.deep.equal([]);
+    expect(transactionReceipt).to.have.property("blockNumber"); // .to.equal("0x");
+    expect(transactionReceipt)
+      .to.have.property("confirmations")
+      .to.equal(transactionResponse.confirmations);
+    expect(transactionReceipt)
+      .to.have.property("cumulativeGasUsed")
+      .to.equal(parseEther("0"));
+    expect(transactionReceipt)
+      .to.have.property("effectiveGasPrice")
+      .to.equal(parseEther("0"));
+    expect(transactionReceipt).to.have.property("byzantium").to.equal(false);
+    expect(transactionReceipt).to.have.property("type").to.equal(2);
+  });
```

Comment:
```suggestion
    // Use .to.deep.includes if this is not full receipt object.
    expect(transactionReceipt).to.deep.equal({
      to: "0x",
      from: "0x",
      //... etc.
    );
  });
```
Should be able to the same throughout tests.